packages/react-jsx-highcharts/src/components/Axis/Axis.js

```diff
@@ -39,7 +39,9 @@ class Axis extends Component {
   }
 
   componentWillUnmount () {
-    this.props.remove();
+    const { remove } = this.props;
+
+    remove && remove();
```

Comment:
I added this check here and in Series.js, because Providers do not return methods when parent objects don't exist (i.e. `axis && axis.remove`), and `componentWillUnmount` could be called after the axis was destroyed with the chart.

I upgraded the `getBoundChartMethod` to return `noop` when method is empty, so this code isn't need any more.